includes/preflight.inc

```diff
@@ -247,6 +232,7 @@ function drush_preflight_site() {
   drush_sitealias_create_self_alias();
 
   // If applicable swaps in shell alias value (or executes it).
+  // TODO: This should probably come later in bootstrap.
```

Comment:
I made that note because the comment above said that drush_shell_alias_replace() might exec shell aliases, and execution should not happen this early in the preflight.  However, it turns out that drush_shell_alias_replace() does not exec here, so this is a good place to call it.  I updated the comment.